src/SurfaceFluxes.jl

```diff
@@ -54,417 +57,752 @@ Surface flux conditions, returned from `surface_conditions`.
+    FT = eltype(sc.L_MO_init)
+    function root_l_mo(x_lmo)
+        root = x_lmo - local_lmo(param_set, x_lmo, sc, scheme)
+        return root
+    end
+    sol = RS.find_zero(
+        root_l_mo,
+        RS.NewtonsMethodAD(sc.L_MO_init),
+        RS.CompactSolution(),
+        RS.SolutionTolerance(sqrt(eps(FT))),
+        100,
+    )
     if sol.converged
-        L_MO, x_star = root_tup[1], SA.SVector(root_tup[2:end])
-        u_star, θ_star = x_star[1], x_star[2]
+        L_MO = sol.root
     else
-        KA.@print("Warning: Unconverged Surface Fluxes\n")
-        L_MO, x_star = root_tup[1], SA.SVector(root_tup[2:end])
-        u_star, θ_star = x_star[1], x_star[2]
+        KA.@print("Warning: Unconverged Surface Fluxes. Returning neutral condition solution \n")
+        L_MO = sol.root
     end
+    return FT(L_MO)
+end
+
+"""
+    obukhov_length(param_set, sc, scheme)
+
+Compute and return the Monin-Obukhov lengthscale (LMO) when surface fluxes
+and friction velocity are known. This is a case-specific implementation of `obukhov_length`;
+iterations to determine LMO are not required.
+"""
+function obukhov_length(param_set, sc::FluxesAndFrictionVelocity, scheme)
+    return -sc.ustar^3 / CPSGS.von_karman_const(param_set) /
+           compute_buoyancy_flux(param_set, sc, scheme)
+end
 
+"""
+  obukhov_length(param_set, sc, scheme)
+Compute and return the Monin-Obukhov lengthscale (LMO) when exchange coefficients are known.
+This is a case-specific implementation of `obukhov_length`;
+iterations to determine LMO are not required.
+"""
+function obukhov_length(param_set, sc::Coefficients, scheme)
+    FT = eltype(sc.Ch)
+    Δθ = TD.dry_pottemp(ts_in(sc)) - TD.dry_pottemp(ts_sfc(sc))
+    ρ_sfc = TD.air_density(sc.state_sfc.ts)
+    cp_m::FT = TD.cp_m(sc.state_in.ts)
+    L_v::FT = TD.latent_heat_vapor(sc.state_in.ts)
+    shf = -ρ_sfc * sc.Ch * cp_m * windspeed(sc) * Δθ
+    lhf = -ρ_sfc * sc.Ch * L_v * windspeed(sc) * Δqt(sc)
+    ustar = sqrt(sc.Cd) * windspeed(sc)
+    intermediate_surface_condition = GivenFluxesAndFrictionVelocity(
+        sc.state_in,
+        sc.state_sfc,
+        shf,
+        lhf,
+        ustar,
+        sc.uf,
+        sc.z0m,
+        sc.z0b,
+    )
+    return obukhov_length(param_set, intermediate_surface_condition, scheme)
+end
+
+function compute_buoyancy_flux(param_set, sc::Coefficients, scheme)
+    FT = eltype(sc.Cd)
+    # Unpack planet parameters
     grav::FT = CPP.grav(param_set)
-    von_karman_const::FT = CPSGS.von_karman_const(param_set)
-    wθ_flux_star = -u_star^3 * θ_scale / (von_karman_const * grav * L_MO)
-    flux = -u_star * x_star
+    ε_vd::FT = CPP.molmass_ratio(param_set)
+    # Compute thermodynamic variables
+    cp_m::FT = TD.cp_m(sc.state_in.ts)
+    L_v::FT = TD.latent_heat_vapor(sc.state_in.ts)
+    # Compute fluxes
+    Δθ = TD.dry_pottemp(ts_in(sc)) - TD.dry_pottemp(ts_sfc(sc))
+    J = -sc.Ch * cp_m * windspeed(sc) * Δθ
```

Comment:
Might be better to be more verbose here, e.g., `shf` and `lhf` instead of `J` and `D` (especially since the `J` flux here differs from the `J` in the design doc).

Reply:
@tapios , here we followed eqs 5.8 and 5.10 in the Design Docs, where if we understood correctly ρJ = SHF and ρD=LHF (with the correction of sensible heat carried by water molecules and potential energy carried by water molecules). Did we miss something in the definition?